### PR TITLE
Attempt to upgrade upb to the latest

### PR DESCRIPTION
**Pull request by veblush**
For the test.

### CODE REVIEW

**Comment by haberman** [comment]
Last fix for this is out for review: https://github.com/protocolbuffers/upb/pull/504

**Comment by haberman** [comment]
Ok the fix is submitted. Want to pull and try again?

**Comment by veblush** [comment]
[grpc_build_artifacts_multiplatform](https://fusion2.corp.google.com/invocations/fc7a122b-2967-4a88-acf4-84e750e2e289): running

**Comment by veblush** [comment]
It still has a UBSAN issue

```
external/upb/upb/decode.c:1069:26: runtime error: null pointer passed as argument 2, which is declared to never be null
/usr/include/string.h:43:28: note: nonnull attribute specified here
    #0 0x8903b0a in upb_Decode /proc/self/cwd/external/upb/upb/decode.c:1069:5
    #1 0x5c4d89a in envoy_extensions_filters_http_router_v3_Router_parse /proc/self/cwd/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/extensions/filters/http/router/v3/router.upb.h:41:7
    #2 0x5c2db94 in grpc_core::(anonymous namespace)::XdsHttpRouterFilter::GenerateFilterConfig(upb_StringView, upb_Arena*) const /proc/self/cwd/src/core/ext/xds/xds_http_filters.cc:43:9
    #3 0x5b7e80f in grpc_core::(anonymous namespace)::HttpConnectionManagerParse(bool, grpc_core::XdsEncodingContext const&, envoy_extensions_filters_network_http_connection_manager_v3_HttpConnectionManager const*, bool, grpc_core::XdsListenerResource::HttpConnectionManager*) /proc/self/cwd/src/core/ext/xds/xds_listener.cc:351:24
    #4 0x5b797c8 in grpc_core::(anonymous namespace)::Ld
```

File: src/core/ext/filters/client_channel/backend_metric.cc [hunk 1]
```diff
@@ -50,10 +50,10 @@ std::map<absl::string_view, double> ParseMap(
 
 const LoadBalancingPolicy::BackendMetricAccessor::BackendMetricData*
 ParseBackendMetricData(const Slice& serialized_load_report, Arena* arena) {
-  upb::Arena upb_arena;
+  upb::Arena upb_Arena;
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/backend_metric.cc:53]: This one should not be changed -- this is a variable name, not a type name.

File: src/core/ext/xds/xds_api.cc [hunk 2]
```diff
@@ -187,12 +187,12 @@ void PopulateBuildVersion(const XdsEncodingContext& context,
                           envoy_config_core_v3_Node* node_msg,
                           const std::string& build_version) {
   std::string encoded_build_version = EncodeStringField(5, build_version);
-  // TODO(roth): This should use upb_msg_addunknown(), but that API is
+  // TODO(roth): This should use upb_Message_AddUnknown(), but that API is
```
Comment (markdroth) [inline src/core/ext/xds/xds_api.cc:190]: Can this now be changed to use `upb_Message_AddUnknown()`?

File: src/core/ext/xds/xds_cluster.cc [hunk 3]
```diff
@@ -275,23 +275,24 @@ grpc_error_handle CdsResourceParse(
         const google_protobuf_Any* typed_config =
             envoy_config_cluster_v3_Cluster_CustomClusterType_typed_config(
                 custom_cluster_type);
-        const upb_strview aggregate_cluster_config_upb_strview =
+        const upb_StringView aggregate_cluster_config_upb_StringView =
```
Comment (markdroth) [inline src/core/ext/xds/xds_cluster.cc:278]: The type should change here, but not the variable name.

File: src/core/ext/xds/xds_api.cc (hunk 2 above)
Comment (nicolasnoble) [reply src/core/ext/xds/xds_api.cc:190]: Good question - @haberman ?

File: src/core/ext/filters/client_channel/backend_metric.cc (hunk 1 above)
Comment (nicolasnoble) [reply src/core/ext/filters/client_channel/backend_metric.cc:53]: Done.

File: src/core/ext/xds/xds_cluster.cc (hunk 3 above)
Comment (nicolasnoble) [reply src/core/ext/xds/xds_cluster.cc:278]: Done.

File: src/core/ext/xds/xds_api.cc (hunk 2 above)
Comment (nicolasnoble) [reply src/core/ext/xds/xds_api.cc:190]: So we'll leave this one for a later, smaller change.

**Review by markdroth — APPROVED** [review]

**Pull request merged**